soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -1203,87 +1203,556 @@ bool Randomizer::IsTrialRequired(RandomizerInf trial) {
+        case RG_LENS_OF_TRUTH:
+            return INV_CONTENT(ITEM_LENS) == ITEM_NONE ? CAN_OBTAIN : CANT_OBTAIN_ALREADY_HAVE;
+        case RG_MAGIC_BEAN:
+        case RG_MAGIC_BEAN_PACK:
+            return BEANS_BOUGHT < 10 ? CAN_OBTAIN : CANT_OBTAIN_ALREADY_HAVE;
+        case RG_MEGATON_HAMMER:
+            return INV_CONTENT(ITEM_HAMMER) == ITEM_NONE ? CAN_OBTAIN : CANT_OBTAIN_ALREADY_HAVE;
+        case RG_FIRE_ARROWS:
+            return INV_CONTENT(ITEM_ARROW_FIRE) == ITEM_NONE ? CAN_OBTAIN : CANT_OBTAIN_ALREADY_HAVE;
+        case RG_ICE_ARROWS:
+            return INV_CONTENT(ITEM_ARROW_ICE) == ITEM_NONE ? CAN_OBTAIN : CANT_OBTAIN_ALREADY_HAVE;
+        case RG_LIGHT_ARROWS:
+            return INV_CONTENT(ITEM_ARROW_LIGHT) == ITEM_NONE ? CAN_OBTAIN : CANT_OBTAIN_ALREADY_HAVE;
+        case RG_DINS_FIRE:
+            return INV_CONTENT(ITEM_DINS_FIRE) == ITEM_NONE ? CAN_OBTAIN : CANT_OBTAIN_ALREADY_HAVE;
+        case RG_FARORES_WIND:
+            return INV_CONTENT(ITEM_FARORES_WIND) == ITEM_NONE ? CAN_OBTAIN : CANT_OBTAIN_ALREADY_HAVE;
+        case RG_NAYRUS_LOVE:
+            return INV_CONTENT(ITEM_NAYRUS_LOVE) == ITEM_NONE ? CAN_OBTAIN : CANT_OBTAIN_ALREADY_HAVE;
+
+        // Bottles
+        case RG_EMPTY_BOTTLE:
+        case RG_BOTTLE_WITH_MILK:
+        case RG_BOTTLE_WITH_RED_POTION:
+        case RG_BOTTLE_WITH_GREEN_POTION:
+        case RG_BOTTLE_WITH_BLUE_POTION:
+        case RG_BOTTLE_WITH_FAIRY:
+        case RG_BOTTLE_WITH_FISH:
+        case RG_BOTTLE_WITH_BLUE_FIRE:
+        case RG_BOTTLE_WITH_BUGS:
+        case RG_BOTTLE_WITH_POE:
+        case RG_RUTOS_LETTER:
+        case RG_BOTTLE_WITH_BIG_POE:
+            return Inventory_HasEmptyBottleSlot() ? CAN_OBTAIN : CANT_OBTAIN_ALREADY_HAVE;
+
+        // Bottle Refills
+        case RG_MILK:
+        case RG_RED_POTION_REFILL:
+        case RG_GREEN_POTION_REFILL:
+        case RG_BLUE_POTION_REFILL:
+        case RG_BUY_FISH:
+        case RG_BUY_RED_POTION_30:
+        case RG_BUY_GREEN_POTION:
+        case RG_BUY_BLUE_POTION:
+        case RG_BUY_BLUE_FIRE:
+        case RG_BUY_BOTTLE_BUG:
+        case RG_BUY_POE:
+        case RG_BUY_FAIRYS_SPIRIT:
+        case RG_BUY_RED_POTION_40:
+        case RG_BUY_RED_POTION_50:
+            return Inventory_HasEmptyBottle() ? CAN_OBTAIN : CANT_OBTAIN_NEED_EMPTY_BOTTLE;
+
+        // Trade Items
+        // TODO: Do we want to be strict about any of this?
+        // case RG_WEIRD_EGG:
+        // case RG_ZELDAS_LETTER:
+        // case RG_POCKET_EGG:
+        // case RG_COJIRO:
+        // case RG_ODD_MUSHROOM:
+        // case RG_ODD_POTION:
+        // case RG_POACHERS_SAW:
+        // case RG_BROKEN_SWORD:
+        // case RG_PRESCRIPTION:
+        // case RG_EYEBALL_FROG:
+        // case RG_EYEDROPS:
+        // case RG_CLAIM_CHECK:
+        // case RG_PROGRESSIVE_GORONSWORD:
+        // case RG_GIANTS_KNIFE:
```

Comment:
To answer the ToDo question, the only strictness I can think of is with the Progressive Biggoron Sword option, but that isn't implemented yet. Everything else doesn't have any requirements I'm aware of besides maybe the child stuff since I don't know if it allows switching back to items before the masks.